src/library/scala/collection/immutable/ChampHashMap.scala

```diff
@@ -436,7 +126,7 @@ final class HashMap[K, +V] private[immutable] (private[immutable] val rootNode:
       case map: HashMap[K, V] =>
         (this eq map) ||
           (this.size == map.size) &&
-            (this.cachedJavaKeySetHashCode == map.cachedJavaKeySetHashCode) &&
+            (this.rootNode equals map.rootNode) &&
```

Comment:
This is wrong! Should be `rootNode.cachedHashCode == map.rootNode.cachedHashCode`